.Lib9c.Tests/Action/Scenario/ItemCraftTest.cs

```diff
@@ -0,0 +1,405 @@
+            Assert.Equal(0, inventoryState.Items.Count);
+
+            // Add materials to inventory
+            stateV2 = CraftUtil.AddMaterialsToInventory(
+                stateV2,
+                _tableSheets,
+                _avatarAddr,
+                allMaterialList,
+                random
+            );
+
+            for (var i = 0; i < recipeList.Count; i++)
+            {
+                // Unlock stage
+                var equipmentRecipe = recipeList[i];
+                stateV2 = CraftUtil.UnlockStage(
+                    stateV2,
+                    _tableSheets,
+                    _worldInformationAddr,
+                    equipmentRecipe.UnlockStage
+                );
+
+                // Do Combination Action
+                var action = new CombinationEquipment
+                {
+                    avatarAddress = _avatarAddr,
+                    slotIndex = i,
+                    recipeId = equipmentRecipe.Id,
+                    subRecipeId = equipmentRecipe.SubRecipeIds?[0],
+                };
+
+                stateV2 = action.Execute(new ActionContext
+                {
+                    PreviousStates = stateV2,
+                    Signer = _agentAddr,
+                    BlockIndex = 0L,
+                    Random = random,
+                });
+                var slotState = stateV2.GetCombinationSlotState(_avatarAddr, i);
+                // TEST: requiredBlock
+                // TODO: Check reduced required block when pet comes in
+                Assert.Equal(equipmentRecipe.RequiredBlockIndex, slotState.RequiredBlockIndex);
+            }
+
+            inventoryState = new Inventory((List)stateV2.GetState(_inventoryAddr));
+            // TEST: Only created equipments should remain in inventory
+            Assert.Equal(recipeList.Count, inventoryState.Items.Count);
+            foreach (var itemId in targetItemIdList)
+            {
+                // TEST: Created equipment should match with targetItemList
+                Assert.NotNull(inventoryState.Items.Where(e => e.item.Id == itemId));
+            }
+        }
+
+        [Theory]
+        [InlineData(1, new[] { 200000 })] // 맛없는 요리
+        public void CraftConsumableTest(int randomSeed, int[] targetItemIdList)
+        {
+            // Disable all quests to prevent contamination by quest reward
+            var (stateV1, stateV2) = QuestUtil.DisableQuestList(
+                _initialStatesWithAvatarStateV1,
+                _initialStatesWithAvatarStateV2,
+                _avatarAddr
+            );
+
+            // Setup requirements
+            var random = new TestRandom(randomSeed);
+            var recipeList = _tableSheets.ConsumableItemRecipeSheet.OrderedList.Where(
+                recipe => targetItemIdList.Contains(recipe.ResultConsumableItemId)
+            ).ToList();
+            var allMaterialList = new List<EquipmentItemSubRecipeSheet.MaterialInfo>();
+            foreach (var recipe in recipeList)
```

Comment:
I think count of the recipes should be checked. (Assert if count is non-zero or positive number)